core/cog-webkit-utils.c

```diff
@@ -98,22 +98,19 @@ format_tls_error (GTlsCertificateFlags errors)
     return g_string_free (str, FALSE);
 }
 
-
 gboolean
-cog_handle_web_view_load_failed_with_tls_errors (WebKitWebView       *web_view,
-                                                 char                *failing_uri,
-                                                 GTlsCertificate     *certificate,
-                                                 GTlsCertificateFlags errors,
-                                                 void                *user_data)
+cog_web_view_load_error_page (WebKitWebView *web_view,
+                              char *failing_uri,
+                              GTlsCertificateFlags errors)
 {
+
     g_autofree char *error_string = format_tls_error (errors);
     return load_error_page (web_view,
                             failing_uri,
                             "TLS Error",
```

Comment:
Now that `cog_web_view_load_error_page()` would be part of the public API, I would rather remove the harcoded page title and instead make it a function parameter.

On a related note, I would only call `format_tls_error()` and add TLS certificate information if the passed `GTlsCertificate` is non-null. Maybe a function argument for an error message would be needed as well :thinking: 

I am not very sure about how to go with this change, it may need a bit more of thinking.